### DISCUSSION TITLE
hii Airflow ,,     i am from mastercard  we looking for airflow which can give access through RBAC (Role based access). and should use our customized python. please help us here.

### DISCUSSION THREAD

**Discussion by kshitij-4567** [apache/airflow#26266] [General]
### Description

hii Airflow ,,     i am from mastercard  we looking for airflow which can give access through RBAC (Role based access). and should use our customized python. please help us here.

### Use case/motivation

we have old airflow url ,  but air flow which can provide RBAC access.

### Related issues

ShortComings :---

a)	User has no option to select the worker node , AirFlow randomly select  the worker node as per the availability.

b)	R-BAC (Role based access control is not available ) for cluster mode, User has a free hand to access others user flow.

c)	Airflow uses it own  Python build.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
PLease never open issues when you want to discuss something (that was clearly explained at the form when you opened an issue).

Your question is - besides - completely unclear. I have no idea what you are asking for and what your problem is - please next time when you post any issue attempt to make your question structured and explanatory. what you created is completly chaotis and impossible to act  on and help you. Invest in clarifying your communication.

But first of all for RBAC controls you can use whatever Flask Application Builder provides. The documentation for you to study starts from here https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html - study it, attempt to use it. Studdy corresponding docuemntation and examples you will find link from there, attempt to realise what you need following the docs and come back when you will not be able to do something by following it, and spending time on understanding how it works.

**Reply by kshitij-4567** [reply]
hii @potiuk ,

Are you from apache airflow ,  
apologies my early question was not in a structured way,

questions :--1 )

we have environment where we provide access to tools using (rbac) role based access, the document which u have shared of user controls 
its seems like user settings where in every time we have defined a user settings , which is not feasible to us.

2)  User  should be to use his worker not dependent on airflow availability.

**Reply by potiuk** [reply]
I am not sure again what you are asking for but likely it is not possible yet. Uou should see https://www.youtube.com/watch?v=nBNbVhz7rhE talk from the Airflow Summit to see what is the state of multi-tenancy and access control. Currently the RBAC ui only applies to UI actions, all the workers actions are executed without any RBAC relations (i.e. task is executed by Airlfow not by the user who triggered it). There is no way to get any authorization "per triggering user" currently - maybe it will be possible in the future but not siin.

**Reply by kshitij-4567** [reply]
let me explain it ,

RBAC (Role back access control),  once the user  raise a access to airflow UI via defined role of airflow,, hosted on one of the linux server, 

why we need roles access , because could user access only their DAG scheduler, instead of having a free hand to all DAGs.

this solution again needs 

admin intervention

DAG-level permissions
For DAG-level permissions exclusively, access can be controlled at the level of all DAGs or individual DAG objects. This includes DAGs.can_create, DAGs.can_read, DAGs.can_edit, and DAGs.can_delete. When these permissions are listed, access is granted to users who either have the listed permission or the same permission for the specific DAG being acted upon. For individual DAGs, the resource name is DAG: + the DAG ID.

For example, if a user is trying to view DAG information for the example_dag_id, and the endpoint requires DAGs.can_read access, access will be granted if the user has either DAGs.can_read or DAG:example_dag_id.can_read access.

**Reply by potiuk** [reply]
No idea what you are asking for

**Reply by kshitij-4567** [reply]
hii @potiuk   please share the documents for the latest  version of Airflow.

1) installation guide
2) steps to install the airflow UI on linux.


we have a older version installed in ur environment 1.10.0

**Reply by potiuk** [reply]
Did you try to read the docs https://airflow.apache.org/ ? Do they have all the answers you look for?
